backend/cmd/migrate/migrations/000005_create_topology_cache_table.up.sql

```diff
@@ -0,0 +1,94 @@
+CREATE OR REPLACE FUNCTION set_timestamp() RETURNS TRIGGER LANGUAGE plpgsql AS '
+BEGIN
+  NEW.updated_at = NOW();
+  RETURN NEW;
+END;
+';
+
+CREATE FUNCTION clean_cache() RETURNS TRIGGER LANGUAGE plpgsql AS '
+BEGIN
+  DELETE FROM topology_cache_data WHERE updated_at <= NOW() - INTERVAL ''60m'';
+  RETURN NULL;
+END;
+';
+
+-- Tables
+CREATE TABLE topology_cache_data (
+  -- id: This is the resource identifier, could be a pod-id, aws instance-id, this value must be unique
+  id varchar UNIQUE,
+  
+  -- id_text_search: This is the same value as the id but in tsquery type for better indexing and fuzzy lookups
+  id_text_search tsquery,
+  
+  -- data: json blob of resource
+  data JSONB,
+  
+  -- resolver_type: is the infrastructure type that the resource belongs to, this could be `kubernetes` or `aws`
+  resolver_type varchar,
```

Comment:
per our earlier discussion, on second thought we will need to add the full type information i'm pretty sure.

if i'm on the pod terminate page and i start type `clutc` i only want to get pod IDs not clusters, but currently it's not possible to restrict it to that, is it?

Reply:
Yeah true will want to have the full resource proto type like the `typeUrl`, let me make some changes.

```
type.googleapis.com/clutch.k8s.v1.Deployment
type.googleapis.com/clutch.k8s.v1.Pod
type.googleapis.com/clutch.aws.ec2.v1.Instance
```